packages/frontend-api/src/Model/Product/Filter/ProductFilterFacade.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Product\Filter;
+
+use Overblog\GraphQLBundle\Definition\Argument;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Model\Category\Category;
+use Shopsys\FrameworkBundle\Model\Product\Brand\Brand;
+use Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterConfig;
+use Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterConfigFactory;
+use Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterData;
+
+class ProductFilterFacade
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Domain\Domain
+     */
+    protected Domain $domain;
+
+    /**
+     * @var \Shopsys\FrontendApiBundle\Model\Product\Filter\ProductFilterDataMapper
+     */
+    protected ProductFilterDataMapper $productFilterDataMapper;
+
+    /**
+     * @var \Shopsys\FrontendApiBundle\Model\Product\Filter\ProductFilterNormalizer
+     */
+    protected ProductFilterNormalizer $productFilterNormalizer;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterConfigFactory
+     */
+    protected ProductFilterConfigFactory $productFilterConfigFactory;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterConfig[]
+     */
+    protected array $productFilterConfigCache = [];
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \Shopsys\FrontendApiBundle\Model\Product\Filter\ProductFilterDataMapper $productFilterDataMapper
+     * @param \Shopsys\FrontendApiBundle\Model\Product\Filter\ProductFilterNormalizer $productFilterNormalizer
+     * @param \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterConfigFactory $productFilterConfigFactory
+     */
+    public function __construct(
+        Domain $domain,
+        ProductFilterDataMapper $productFilterDataMapper,
+        ProductFilterNormalizer $productFilterNormalizer,
+        ProductFilterConfigFactory $productFilterConfigFactory
+    ) {
+        $this->productFilterDataMapper = $productFilterDataMapper;
+        $this->productFilterNormalizer = $productFilterNormalizer;
+        $this->productFilterConfigFactory = $productFilterConfigFactory;
+        $this->domain = $domain;
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterConfig
+     */
+    public function getProductFilterConfigForAll(): ProductFilterConfig
+    {
+        $cacheKey = 'all';
```

Comment:
_thought_

What do you think about using Symfony cache with ArrayAdapter(memory)?
I have two advantages in mind:
 - responsibilities will be split (this class now act as a cache and as a creator)
 - when necessary, it would be really easy to start caching filters in some persistent storage (redis)

Reply:
Well we have few cached facades and I am just storing current value I have, so I do not load it again during one run. That ArrayAdapter seems ok - but not sure if it is necessary. 

So the questions are - do you think, this is important? Should we use it in all cached facades?

Reply:
Not sure if in all cached facades. 
This filter config seems like something I want to store in some persistent cache as it's valid until some product changes.
The only reason I don't suggest to use redis cache is I don't want to solve cache invalidation in this PR. 

so I suggest using cacheinterface with arrayadapter because I believe it could be beneficial in the future.
But I don't force it as the current implementation also make sense and works as expected.

Reply:
I am ok with current solution, lets discuss how to do this in all around the app.